src/plugins/observables.js

```diff
@@ -1,9 +1,40 @@
 /* @flow */
 import isObservable from 'is-observable'
 import StyleRule from '../rules/StyleRule'
-import type {Rule} from '../types'
+import createRule from '../utils/createRule'
+import type {Observable, Rule, RuleOptions, JssStyle} from '../types'
 
 export default {
+  onCreateRule(name: string, decl: JssStyle, options: RuleOptions): Rule | null {
+    if (isObservable(decl)) {
+      // Cast `decl` to `Observable`, since it passed the type guard
+      const props$ = (decl: Observable<{[string]: string}>);
+
+      // We know `rule` is a `StyleRule`, and the other types don't have a
+      // `prop` method, so we must explicitly cast to `StyleRule`
+      const rule = ((createRule(name, {}, options): any): StyleRule);
+
+      // `stream.subscribe()` returns a subscription, which should be explicitly
+      // unsubscribed from when we know this sheet is no longer needed, but I
+      // don't see any hooks to do that in the plugin API.  The Observable props
+      // implementation doesn't store its subscription either, so this is a TODO
+      // for both.
+      props$.subscribe(
+        (props) => {
+          Object.entries(props).forEach(
```

Comment:
This could be a performance sensitive place, I normally try to avoid function calls and function generations as far as I can. I would simply use a for loop, because we expect only plain objects here